pkg/server/image_pull_test.go

```diff
@@ -103,16 +105,50 @@ func TestParseAuth(t *testing.T) {
 	}
 }
 
-func TestAddDefaultEndpoint(t *testing.T) {
+func TestRegistryEndpoints(t *testing.T) {
 	for desc, test := range map[string]struct {
-		endpoints []string
-		host      string
-		expected  []string
+		mirrors  map[string]criconfig.Mirror
+		host     string
+		expected []string
 	}{
-		"default endpoint not in list": {
-			endpoints: []string{
+		"no mirror configured": {
+			mirrors: map[string]criconfig.Mirror{
+				"registry-1.io": {
+					Endpoints: []string{
+						"https://registry-1.io",
+						"https://registry-2.io",
+					},
+				},
+			},
+			host: "registry-3.io",
+			expected: []string{
+				"https://registry-3.io",
+			},
+		},
+		"wildcard mirror configured": {
+			mirrors: map[string]criconfig.Mirror{
+				"*": {
+					Endpoints: []string{
+						"https://registry-1.io",
+						"https://registry-2.io",
+					},
+				},
+			},
+			host: "registry-3.io",
+			expected: []string{
```

Comment:
needs a couples tests with host and wildcard specified...where host matches and doesn't to confirm ordering

Reply:
Good point.

Reply:
Done.